core/src/sumeragi/mod.rs

```diff
@@ -250,9 +251,38 @@ impl Sumeragi {
     pub fn initialize_and_start_thread(
         sumeragi: Arc<Self>,
         genesis_network: Option<GenesisNetwork>,
+        block_hashes: Vec<HashOf<VersionedCommittedBlock>>,
     ) -> ThreadHandler {
         let wsv = sumeragi.wsv_mutex_access().clone();
 
+        for (i, block_hash) in block_hashes
+            .iter()
+            .take(block_hashes.len().saturating_sub(1))
+            .enumerate()
```

Comment:
I would suggest the following to avoid conversion: 

```suggestion
            .zip(1u64..)
```

Reply:
sure